catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlDatabaseOperations.java

```diff
@@ -47,11 +47,30 @@ public String generateCreateDatabaseSql(
 
   @Override
   public String generateDropDatabaseSql(String databaseName, boolean cascade) {
+    final String dropDatabaseSql = "DROP DATABASE `" + databaseName + "`";
     if (cascade) {
-      throw new UnsupportedOperationException(
-          "MySQL does not support CASCADE option for DROP DATABASE.");
+      return dropDatabaseSql;
     }
-    return "DROP DATABASE `" + databaseName + "`";
+
+    try (final Connection connection = this.dataSource.getConnection()) {
+      String query = "SELECT * FROM information_schema.TABLES WHERE TABLE_SCHEMA = ?";
```

Comment:
I prefer using normal sql to query tables instead of querying internal table.

Reply:
You mean we can use the `show tables in db_name` sentence?

Reply:
yes

Reply:
Fixed.